router/usecase/router_usecase.go

```diff
@@ -511,14 +514,36 @@ func (r *routerUseCaseImpl) GetCustomDirectQuoteMultiPool(ctx context.Context, t
 			return nil, err
 		}
 
+		route := quote.GetRoute()
+		if len(route) != 1 {
+			return nil, fmt.Errorf("custom direct quote must have 1 route, had: %d", len(route))
+		}
+
+		poolsInRoute := route[0].GetPools()
+		if len(poolsInRoute) != 1 {
+			return nil, fmt.Errorf("custom direct quote route must have 1 pool, had: %d", len(poolsInRoute))
+		}
+
 		// the amountOut value is the amount out of last the tokenOutDenom
 		result.AmountOut = quote.GetAmountOut()
 
 		// append each pool to the route
-		result.Route = append(result.Route, quote.GetRoute()...)
+		pools = append(pools, poolsInRoute...)
```

Comment:
I am not sure I fully follow what this code does. I think multi route issue was resolved by by requesting "single" route quotes in the tests. How does limiting to a single pool resolves an issue?

Reply:
I have opened proposal PR to be merged into this one: https://github.com/osmosis-labs/sqs/pull/456

Reply:
Let's align on route definitions. There is a "multi hop route" and there are "split routes".

Multi hop route is a single route over multiple pools.

Split routes are splits over multiple routes in proportions of the original amount in.

`/custom-direct-quote` only supports "multi-hop routes". Split routes are not supported.

> However I am not following how limiting to a single pool resolves the issue?

What was done in this PR is to ensure that we construct a "multi hop route" from separate quote responses to a single pool route.

> Since routes were returned as split-routes

The routes are returned as a slice of a split route interface but those might not even be split routes under the hood. The naming of the interface seems to be chosen poorly and causes confusion.

Let me know if any follow up thoughts on this 🙏